overlord/ifacestate/handlers.go

```diff
@@ -427,3 +427,70 @@ func (m *InterfaceManager) doDisconnect(task *state.Task, _ *tomb.Tomb) error {
 	setConns(st, conns)
 	return nil
 }
+
+// transitionConnectionsCoreMigration will transition all connections
+// from oldName to newName. Note that this is only useful when you
+// know that newName supports everything that oldName supports,
+// otherwise you will be in a world of pain.
+func (m *InterfaceManager) transitionConnectionsCoreMigration(st *state.State, oldName, newName string) error {
+	// transition over, ubuntu-core has only slots
+	conns, err := getConns(st)
+	if err != nil {
+		return err
+	}
+
+	for id := range conns {
+		connRef := interfaces.ConnRef{}
+		if err := connRef.ParseID(id); err != nil {
+			return err
+		}
+		if connRef.SlotRef.Snap == oldName {
+			connRef.SlotRef.Snap = newName
+			conns[connRef.ID()] = conns[id]
+			delete(conns, id)
+		}
+	}
+	setConns(st, conns)
+
+	if err := m.reloadConnections(oldName); err != nil {
```

Comment:
Thanks, this is super interesting! 

The branch is indeed assuming that we don't need to re-generate security profiles because core/ubuntu-core are absolutely identical and nothing in the generated policies has the slot-name (core or ubuntu-core) encoded. I went over the interfaces listed in `snap/implicit.go` and looked at the rule generating code. Some interfaces generate udev rules that encode the name of the plug (like `iio,physical-memory-{control,observe}`) but not the name of the slot AFAICT.

The network-manager/upower-observe interfaces are using `###SLOT_SECURITY_TAG###`  but that is translated to "unconfined" in classic. Modem-manager has similar code but its not setting `###SLOT_SECURITY_TAG###` to unconfined (which might be a bug?).

The gpio interface is using the slot name as the service name, but that is not a concern to us.

I think we are good here but please do let me know if you can think of anything I might have missed.